firetweet/src/main/java/org/getlantern/firetweet/activity/support/UpdaterActivity.java

```diff
@@ -0,0 +1,189 @@
+
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+
+        btn = (Button)findViewById(R.id.install_update);
+        btn.setTypeface(font);
+        btn.setOnClickListener(new OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                runUpdater();
+            }
+        });
+    }
+
+    private void runUpdater() {
+        updateIcon.setVisibility(View.GONE);
+        spinner.setVisibility(View.VISIBLE);
+
+        String[] updaterParams = {APK_URL};
+        mUpdaterTask = new UpdaterTask(this);
+        mUpdaterTask.execute(updaterParams);
+    }
+
+    public void dismissActivity() {
+
+        finish();
+    }
+
+    @Override
+    public void finish() {
+        super.finish();
+    }
+
+    static class UpdaterTask extends AsyncTask<String, String, String> {
+
+        private final UpdaterActivity mActivity;
+        private final Context context;
+
+        private static final String APK_PATH = "/sdcard/FireTweet.apk";
+
+        UpdaterTask(final UpdaterActivity activity) {
+            mActivity = activity;
+            context = mActivity.getApplicationContext();
+        }
+
+        @Override
+        protected String doInBackground(String... sUrl) {
+            String path = APK_PATH;
+            try {
+                Log.d(LOG_TAG, "Attempting to download new APK from " + sUrl[0]);
+                URL url = new URL(sUrl[0]);
+                URLConnection connection = url.openConnection();
+                connection.connect();
+
+                int fileLength = connection.getContentLength();
+
+                // download the file
+                InputStream input = new BufferedInputStream(url.openStream());
+                OutputStream output = new FileOutputStream(path);
+
+                byte data[] = new byte[1024];
+                long total = 0;
+                int count;
+                while ((count = input.read(data)) != -1) {
+                    total += count;
+                    int progress = (int) (total * 100 / fileLength);
+                    publishProgress(Integer.toString(progress));
+                    output.write(data, 0, count);
+                }
+
+                output.flush();
+                output.close();
+                input.close();
+            } catch (Exception e) {
+                Log.e(LOG_TAG, "Error installing new APK..");
+                Log.e(LOG_TAG, e.getMessage());
+                displayInstallError();
+            }
+            return path;
+        }
+
+        // show an alert when the update fails
+        // and mention where the user can download the latest version
+        // this also dismisses the current updater activity
+        protected void displayInstallError() {
+            AlertDialog alertDialog = new AlertDialog.Builder(mActivity).create();
+            alertDialog.setTitle("Error Downloading Update");
```

Comment:
Yes, thanks. I added the update text to ``strings.xml`` and will push them to Transifex